desktop/core/src/desktop/js/onePageViewModel.js

```diff
@@ -446,6 +446,12 @@ class OnePageViewModel {
       self.loadApp('importer');
       self.getActiveAppViewModel(vm => {
         vm.createWizard.source.path(path);
+        vm.currentStep(2);
+        vm.createWizard.source.interpreter('hive');
+        vm.createWizard.destination.name('default.xyz1');
+        window.setTimeout(() => {
+          vm.createWizard.indexFile();
+        }, 1000);
```

Comment:
Why 1000? Does 0 work?

Reply:
No, 0 is not working. needs some time to execute the above functions (vm.createWizard.source.path(path) trigger some function in the importer.mako so need some waiting time to execute all of them).
https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/templates/importer.mako#L1811
@JohanAhlen

Reply:
Right, so you'd need to find another way. 1 second is arbitrary and won't guarantee that it works in 100% of the cases, are you waiting for an API call? What if it takes longer than 1 second?

One option is to use huePubSub and publish an event when whatever you're waiting for is done. Then you can do huePubSub.subscribeOnce here.